src/Abuse/Adapters/TimeLimit/Pool.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Utopia\Abuse\Adapters\TimeLimit;
+
+use Utopia\Abuse\Adapters\TimeLimit;
+use Utopia\Pools\Pool as UtopiaPool;
+use Utopia\Abuse\Adapters\TimeLimit\Redis as RedisAdapter;
+use Redis;
+
+class Pool extends TimeLimit
+{
+    /**
+    * @var UtopiaPool<covariant Redis>
+    */
+    protected UtopiaPool $pool;
+
+    protected string $key;
+    protected int $limit;
+    protected int $seconds;
+
+    /**
+     * @param string $key
+     * @param int $limit
+     * @param int $seconds
+     * @param  UtopiaPool<covariant Redis>  $pool The pool to use for connections. Must contain instances of TimeLimit.
+     *
+     * @throws \Exception
+     */
+    public function __construct(string $key, int $limit, int $seconds, UtopiaPool $pool)
+    {
+        $this->pool = $pool;
+        $this->key = $key;
+        $this->limit = $limit;
+        $this->seconds = $seconds;
+
+        $this->pool->use(function (mixed $resource) {
+            if (! ($resource instanceof Redis)) {
+                throw new \Exception('Pool must contain instances of '.Redis::class);
+            }
+        });
+    }
+
+    /**
+     * Forward method calls to the internal adapter instance via the pool.
+     *
+     * Required because __call() can't be used to implement abstract methods.
+     *
+     * @param  string  $method
+     * @param  array<mixed>  $args
+     * @return mixed
+     */
+    public function delegate(string $method, array $args): mixed
+    {
+        return $this->pool->use(function (Redis $redis) use ($method, $args) {
+            $adapter = new RedisAdapter($this->key, $this->limit, $this->seconds, $redis);
```

Comment:
Can we instantiate the adapter in the pool init function instead? This way, we instantiate a new adapter for every call

Reply:
I did not see a clear path to implement it that way in the init function creating a pool of timeLimit :(